src/System.Data.Odbc/src/System/Data/Odbc/OdbcParameter.cs

```diff
@@ -747,26 +747,31 @@ internal void PrepareForBind(OdbcCommand command, short ordinal, ref int paramet
                 }
             };
 
-            int cbParameterSize = GetParameterSize(value, offset, ordinal);      // count of bytes for the data, for SQLBindParameter
+            int cbParameterSize = GetParameterSize(value, offset, ordinal); // count of bytes for the data, for SQLBindParameter
 
-            // here we upgrade the datatypes if the given values size is bigger than the types columnsize
-            //
+            // Upgrade input value type if the size of input value is bigger than the max size of the input value type.
             switch (_bindtype._sql_type)
             {
-                case ODBC32.SQL_TYPE.VARBINARY: // MDAC 74372
-                    // Note: per definition DbType.Binary does not support more than 8000 bytes so we change the type for binding
-                    if ((cbParameterSize > 8000))
-                    { _bindtype = TypeMap._Image; } // will change to LONGVARBINARY
+                case ODBC32.SQL_TYPE.VARBINARY:
+                    // Max length of VARBINARY is 8,000 of byte array.
+                    if (_internalSize > 8000)
```

Comment:
Should the local "size" variable be used instead of _internalSize? "size" is originally set to the value of _internalSize, but there's some extra logic for adjusting the size value earlier in this method.

Reply:
`_internalSize` represents output string length, and it is used latest before hitting this code.
`_internalSize` is used in `GetParameterSize(value, offset, ordinal);` method for calculating output string buffer size. 
https://github.com/dotnet/corefx/blob/e7723074a7ee7cfbc1da516108d2ad85082d77d4/src/System.Data.Odbc/src/System/Data/Odbc/OdbcParameter.cs#L461-L531
It seems `_internalSize` is more accurate.